gobblin-data-management/src/main/java/gobblin/data/management/copy/replication/WatermarkMetadataGenerationCommitStep.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.data.management.copy.replication;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FSDataOutputStream;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+
+import com.google.common.base.Charsets;
+
+import gobblin.commit.CommitStep;
+
+/**
+ * A {@link CommitStep} to write watermark metadata to Hdfs
+ * @author mitu
+ *
+ */
+public class WatermarkMetadataGenerationCommitStep implements CommitStep {
+  private final String fsUriString;
+  private final Path targetDirPath;
```

Comment:
Did you verify `Path` is serialized and deserialized correctly? I seem to remember having issues with this and having to store the `String` instead.